src/node-explorer-provider.ts

```diff
@@ -56,12 +58,81 @@ export class NodeExplorerProvider
     this.registerRefresh();
     this.registerOpenDocsLink();
     this.registerDownloadCommand();
+    this.pollForUpdates();
   }
 
   getTreeItem(element: PeerBaseTreeItem): vscode.TreeItem {
     return element;
   }
 
+  private async getPeers() {
+    this.previousStatus = this.currrentStatus;
+    this.currrentStatus = await this.ts.getPeers();
+    return this.currrentStatus;
+  }
+
+  private async diffRelay() {
+    const status = await this.getPeers();
+    const prevStatus = this.previousStatus;
+
+    if (prevStatus) {
+      if (status.Errors) {
+        if (!prevStatus.Errors || !status.Errors.every((e, i) => e === prevStatus.Errors?.[i])) {
+          return true;
+        }
+      }
+
+      if (status.CurrentTailnet.Name !== prevStatus.CurrentTailnet.Name) {
+        return true;
+      }
+
+      if (status.CurrentTailnet.MagicDNSEnabled !== prevStatus.CurrentTailnet.MagicDNSEnabled) {
+        return true;
+      }
+
+      if (status.CurrentTailnet.MagicDNSSuffix !== prevStatus.CurrentTailnet.MagicDNSSuffix) {
+        return true;
+      }
+
+      for (let i = 0; i < status.PeerGroups.length; ++i) {
```

Comment:
@marwan-at-work, do you know if this order is consistent? I remember you doing some digging related to that.